spec/rubocop/cop/style/empty_literal_spec.rb

```diff
@@ -2,37 +2,26 @@
 
 RSpec.describe RuboCop::Cop::Style::EmptyLiteral, :config do
   describe 'Empty Array' do
-    it 'registers an offense for Array.new()' do
-      expect_offense(<<~RUBY)
-        test = Array.new()
-               ^^^^^^^^^^^ Use array literal `[]` instead of `Array.new`.
-      RUBY
-
-      expect_correction(<<~RUBY)
-        test = []
-      RUBY
-    end
-
-    it 'registers an offense for Array.new' do
-      expect_offense(<<~RUBY)
-        test = Array.new
-               ^^^^^^^^^ Use array literal `[]` instead of `Array.new`.
-      RUBY
+    shared_examples 'registers_and_corrects' do |initializer:|
+      it "registers an offense for #{initializer}" do
+        expect_offense(<<~RUBY)
+          test = #{initializer}
+                 #{'^' * initializer.length} Use array literal `[]` instead of `#{initializer}`.
+        RUBY
 
-      expect_correction(<<~RUBY)
-        test = []
-      RUBY
+        expect_correction(<<~RUBY)
+          test = []
+        RUBY
+      end
```

Comment:
`shared_examples` don't always improve readability and reproducibility. Could you expand them into individual tests, as in the original specs?